pkg/server/admin.go

```diff
@@ -2978,122 +2993,71 @@ func (s *adminServer) dataDistributionHelper(
 
-		// Insert database if it doesn't exist.
-		dbInfo, ok := resp.DatabaseInfo[*dbName]
 		if !ok {
-			dbInfo = serverpb.DataDistributionResponse_DatabaseInfo{
-				TableInfo: make(map[string]serverpb.DataDistributionResponse_TableInfo),
+			tableName := (*string)(row[1].(*tree.DString))
+			schemaName := (*string)(row[2].(*tree.DString))
+			fqTableName := fmt.Sprintf("%s.%s", tree.NameStringP(schemaName), tree.NameStringP(tableName))
+			dbName := (*string)(row[3].(*tree.DString))
+
+			// Look at whether it was dropped.
+			var droppedAtTime *time.Time
+			droppedAtDatum, ok := row[4].(*tree.DTimestamp)
+			if ok {
+				droppedAtTime = &droppedAtDatum.Time
 			}
-			resp.DatabaseInfo[*dbName] = dbInfo
-		}
-
-		// Get zone config for table.
-		zcID := int64(0)
 
-		if droppedAtTime == nil {
-			// TODO(vilterp): figure out a way to get zone configs for tables that are dropped
-			zoneConfigQuery := fmt.Sprintf(
-				`SELECT zone_id FROM [SHOW ZONE CONFIGURATION FOR TABLE %s.%s.%s]`,
-				(*tree.Name)(dbName), (*tree.Name)(schemaName), (*tree.Name)(tableName),
-			)
-			row, err := s.internalExecutor.QueryRowEx(
-				ctx, "admin-replica-matrix", nil, /* txn */
-				sessiondata.InternalExecutorOverride{User: userName},
-				zoneConfigQuery,
-			)
-			if err != nil {
-				return nil, err
-			}
-			if row == nil {
-				return nil, errors.Errorf(
-					"could not get zone config for table %s; 0 rows returned", *tableName,
-				)
+			// Insert database if it doesn't exist.
+			dbInfo, ok := resp.DatabaseInfo[*dbName]
+			if !ok {
+				dbInfo = serverpb.DataDistributionResponse_DatabaseInfo{
+					TableInfo: make(map[string]serverpb.DataDistributionResponse_TableInfo),
+				}
+				resp.DatabaseInfo[*dbName] = dbInfo
 			}
 
-			zcID = int64(tree.MustBeDInt(row[0]))
-		}
-
-		// Insert table.
-		tableInfo := serverpb.DataDistributionResponse_TableInfo{
-			ReplicaCountByNodeId: make(map[roachpb.NodeID]int64),
-			ZoneConfigId:         zcID,
-			DroppedAt:            droppedAtTime,
-		}
-		dbInfo.TableInfo[fqTableName] = tableInfo
-		tableInfosByTableID[tableID] = tableInfo
-	}
-	if err != nil {
-		return nil, err
-	}
-
-	// Get replica counts.
-	if err := s.db.Txn(ctx, func(txnCtx context.Context, txn *kv.Txn) error {
-		acct := s.memMonitor.MakeBoundAccount()
-		defer acct.Close(txnCtx)
+			// Get zone config for table.
+			zcID := int64(0)
 
-		kvs, err := kvclient.ScanMetaKVs(ctx, txn, roachpb.Span{
-			Key:    keys.SystemSQLCodec.TablePrefix(keys.MaxReservedDescID + 1),
-			EndKey: keys.MaxKey,
-		})
-		if err != nil {
-			return err
-		}
+			if droppedAtTime == nil {
+				// TODO(vilterp): figure out a way to get zone configs for tables that are dropped
+				zoneConfigQuery := fmt.Sprintf(
+					`SELECT zone_id FROM [SHOW ZONE CONFIGURATION FOR TABLE %s.%s.%s]`,
+					(*tree.Name)(dbName), (*tree.Name)(schemaName), (*tree.Name)(tableName),
+				)
+				row2, err := s.internalExecutor.QueryRowEx(
+					ctx, "data-distribution", nil, /* txn */
```

Comment:
Existing code that you aren't responsible for fixing, but..a potential quick win follow up:

- `SHOW ZONE CONFIGURATION` doesn't have a column named zone_id unless I just missed it.
- If that is true, this is always returning 0.
- That's not a huge problem because, as far as I can tell via grep, no one is reading this column?

So maybe we can remove this inner query complete?